modules/services/recoll.nix

```
@@ -0,0 +1,170 @@
+  };
+
+  # The actual object we're going to use for this module. This is for the sake
+  # of consistency (and dogfooding the settings format implementation).
+  settingsFormat = recollConfFormat { };
+in {
+  meta.maintainers = [ maintainers.foo-dogsquared ];
+
+  options.services.recoll = {
+    enable = mkEnableOption "Recoll file index service";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.recoll;
+      defaultText = literalExpression "pkgs.recoll";
+      description = ''
+        Package providing the <literal>recoll</literal> binary.
+      '';
+      example = literalExpression "(pkgs.recoll.override { withGui = false; })";
+    };
+
+    startAt = mkOption {
+      type = types.str;
+      default = "hourly";
+      example = "00/2:00";
+      description = ''
+        When or how often the periodic update should run. Must be the format
+        described from
+        <citerefentry>
+          <refentrytitle>systemd.time</refentrytitle>
+          <manvolnum>7</manvolnum>
+        </citerefentry>.
+      '';
+    };
+
+    settings = mkOption {
+      type = settingsFormat.type;
+      default = { };
+      description = ''
+        The configuration to be written at
+        <filename>''${config.services.recoll.configDir}/recoll.conf</filename>.
+
+        See
+        <citerefentry>
+          <refentrytitle>recoll</refentrytitle>
+          <manvolnum>5</manvolnum>
+        </citerefentry> for more details about the configuration.
+      '';
+      example = literalExpression ''
+        {
+          topdirs = "~/Downloads ~/Documents ~/projects";
+
+          "~/Downloads" = {
+            "skippedNames+" = "*.iso";
+          };
+
+          "~/projects" = {
+            "skippedNames+" = "node_modules target result";
+          };
+        }
+      '';
+    };
+
+    configDir = mkOption {
+      type = types.str;
+      default = "~/.recoll";
+      example = literalExpression "\${config.xdg.configHome}/recoll";
+      description = ''
+        The directory to contain Recoll configuration files. This will be set
+        as <literal>RECOLL_CONFDIR</literal>.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      (lib.hm.assertions.assertPlatform "services.recoll" pkgs
+        lib.platforms.linux)
+    ];
+
+    home.packages = [ cfg.package ];
+
+    home.sessionVariables = { RECOLL_CONFDIR = cfg.configDir; };
+
+    home.file."${cfg.configDir}/recoll.conf".source =
+      settingsFormat.generate "recoll-${config.home.username}" cfg.settings;
+
+    systemd.user.services.recollindex = {
+      Unit = {
+        Description = "Recoll index update";
+        Documentation = [
+          "man:recoll"
+          "man:recollindex"
+          "https://www.lesbonscomptes.com/recoll/usermanual/"
+        ];
+      };
+
+      Service = {
+        ExecStart = "${cfg.package}/bin/recollindex";
+        Environment = [ "RECOLL_CONFDIR=${cfg.configDir}" ];
```

To avoid misunderstandings if the path contains a space or something:

```suggestion
        Environment = [ "RECOLL_CONFDIR=${escapeShellArg cfg.configDir}" ];
```